runhouse/resources/hardware/launcher_utils.py

```diff
@@ -113,13 +113,19 @@ class DenLauncher(Launcher):
 
     @classmethod
     def _update_from_den_response(cls, cluster, config: dict):
-        """Updates cluster with config from Den."""
-        cluster.launched_properties = config.get("launched_properties", {})
-        cluster.ips = config.get("ips", {})
-        cluster.stable_internal_external_ips = config.get(
-            "stable_internal_external_ips", {}
-        )
-        cluster.ssh_properties = config.get("ssh_properties", {})
+        """Updates cluster with config from Den. Only add fields if found."""
```

Comment:
fwiw i think client port might need to be updated too, at least in the diff that Paul showed us it was missing in the metadata after calling `up_if_not()`

Reply:
oh yeah, good call